### PR TITLE
Use fnmatch for callMatches

### PR DESCRIPTION
**Pull request by ruudk**
I think it's good to always use `fnmatch` instead of doing strpos * trick.

### CODE REVIEW

**Comment by spaze** [comment]
Please add an example of what benefits it brings, and tests. Thanks.

**Comment by ruudk** [comment]
Rebased the PR. Please review this commit by commit as it explains what is done and why. 

If you accept this, you can squash merge it if that's what you want.

File: src/DisallowedHelper.php [hunk 1]
```diff
@@ -143,7 +143,7 @@ public function createConstantsFromConfig(array $config): array
 	public function getDisallowedMessage(?Node $node, Scope $scope, string $name, ?string $displayName, array $disallowedCalls): array
 	{
 		foreach ($disallowedCalls as $disallowedCall) {
-			if ($this->callMatches($disallowedCall, $name) && !$this->isAllowed($scope, $node, $disallowedCall)) {
+			if ($this->matches($disallowedCall->getCall(), $name) && !$this->isAllowed($scope, $node, $disallowedCall)) {
```
Comment (spaze) [inline src/DisallowedHelper.php:146]: No need to modify the line.

**Review by spaze — CHANGES_REQUESTED** [review]
I'm not sure this is useful but am fine merging it anyway. Though please keep the diff as small as possible.

File: src/DisallowedHelper.php (hunk 1 above)
Comment (ruudk) [reply src/DisallowedHelper.php]: Reverted.

**Comment by ruudk** [comment]
@spaze Updated the PR, it's now ready ✅

**Review by spaze — APPROVED** [review]

**Comment by spaze** [comment]
Awesome, thank you. Also thanks for the detailed commit descriptions, makes a reviewer's life much easier significantly increasing the chance it will get merged (soon, or at all) 👍

**Comment by spaze** [comment]
Just released this in 1.3.0, thanks.

**Pull request merged**